backend/src/main/java/hr/fer/tzk/rankup/controller/MemberController.java

```diff
@@ -44,131 +47,66 @@ public ResponseEntity<DetailedMemberDto> findMember(@PathVariable Long idMember)
-//            return ResponseEntity.badRequest().body("Missing email");
-//        }
-//
-//        return null;
-//    }
-
-//    @PostMapping
-//    public ResponseEntity<String> createMember(@RequestBody MemberNotRegisteredDto memberDto) throws URISyntaxException {
-//        ResponseEntity<String> validationResponse = validateMemberDto(memberDto);
-//        if (validationResponse != null) {
-//            return validationResponse;
-//        }
-//
-//        Optional<Member> memberWithSameJmbag = memberService.findMemberByJmbag(memberDto.getJmbag());
-//        if (memberWithSameJmbag.isPresent()) {
-//            return ResponseEntity.status(HttpStatus.CONFLICT).body("JMBAG already in use");
-//        }
-//
-//        Member newMember;
-//        try {
-//            newMember = new Member(memberDto.getFirstName(), memberDto.getLastName(), memberDto.getJmbag());
-//        } catch (IllegalArgumentException exception) {
-//            return ResponseEntity.badRequest().body(exception.getMessage());
-//        }
-//        memberService.createMember(newMember);
-//
-//        Optional<Member> createdMemberOpt = memberService.findMemberByJmbag(memberDto.getJmbag());
-//        if (createdMemberOpt.isEmpty()) {
-//            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("Error creating member");
-//        }
-//
-//        Long createdMemberId = createdMemberOpt.get().getId();
-//        URI location = new URI("/api/members/" + createdMemberId);
-//
-//        return ResponseEntity.created(location).build();
-//    }
-//
-//    @PutMapping("/{idMember}")
-//    public ResponseEntity<String> updateMember(@PathVariable Long idMember, @RequestBody MemberGlobalDto memberDto) {
-//        ResponseEntity<String> validationResponse = validateMemberGlobalDto(memberDto);
-//        if (validationResponse != null) {
-//            return validationResponse;
-//        }
-//
-//        Optional<Member> memberOpt = memberService.findMemberById(idMember);
-//        if (memberOpt.isEmpty()) {
-//            return ResponseEntity.notFound().build();
-//        }
-//
+    @PostMapping
+    public ResponseEntity<String> createMember(@Valid @RequestBody BasicMemberForm member) throws URISyntaxException {
+        Optional<Member> memberWithSameJmbag = memberService.findMemberByJmbag(member.getJmbag());
+        if (memberWithSameJmbag.isPresent()) {
+            return ResponseEntity.status(HttpStatus.CONFLICT).body("JMBAG already in use");
+        }
+
+        Member newMember;
+        try {
+            newMember = new Member();
+            newMember.setFirstName(member.getFirstName());
+            newMember.setLastName(member.getLastName());
+            newMember.setJmbag(member.getJmbag());
+        } catch (IllegalArgumentException exception) {
+            return ResponseEntity.badRequest().body(exception.getMessage());
+        }
+        memberService.createMember(newMember);
```

Comment:
Možda bolje da prebacimo u service ovo? Da kontroler bude minimalan što se tiče nekih provjera i postavljanja atributa

Reply:
Moze